### PR TITLE
add nginx template

### PR DESCRIPTION
**Pull request by ChoiByungWook**
Going to add unit tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: etc/nginx.conf.template [hunk 1]
```diff
@@ -19,7 +19,7 @@ http {
   }
 
   server {
-    listen 8080 deferred;
+    listen %NGINX_HTTP_PORT% deferred;
```
Comment (orchidmajumder) [inline etc/nginx.conf.template]: who is going to provide this variable?

Comment (ChoiByungWook) [reply etc/nginx.conf.template]: changes below should do that. ahahah forgot to add that commit.

File: etc/nginx.conf.template [hunk 2]
```diff
@@ -34,5 +34,6 @@ http {
     location / {
       return 404 "{}";
     }
+
   }
-}
+}
```
Comment (jesterhazy) [inline etc/nginx.conf.template]: EOF newline

File: src/sagemaker_containers/_server.py [hunk 3]
```diff
@@ -13,16 +13,36 @@
...
+nginx_config_file = pkg_resources.resource_filename(sagemaker_containers.__name__, '/etc/nginx.conf')
+nginx_config_template_file = pkg_resources.resource_filename(sagemaker_containers.__name__, '/etc/nginx.conf.template')
+
+
+def _create_nginx_config(serving_env):
+    template = _files.read_file(nginx_config_template_file)
+
+    pattern = re.compile(r'%(\w+)%')
+    template_values = {
+        'NGINX_HTTP_PORT': serving_env.port
```
Comment (jesterhazy) [inline src/sagemaker_containers/_server.py]: serving_env.port needs better name... http_port

File: src/sagemaker_containers/_env.py [hunk 4]
```diff
@@ -819,3 +823,10 @@ def default_accept(self):  # type: () -> str
             str: The desired MIME type of the inference in the response. For example: application/json.
                 Default: application/json"""
         return self._default_accept
+
+    @property
+    def port(self):  # type: () -> str
```
Comment (jesterhazy) [inline src/sagemaker_containers/_env.py]: needs better name... http_port or bind_port

File: src/sagemaker_containers/_env.py [hunk 5]
```diff
@@ -819,3 +823,10 @@ def default_accept(self):  # type: () -> str
...
+        """Returns:
+            str: Port that SageMaker will use to handle invocations and pings against the running
+                Docker container. Default should be 8080. For example: 8080"""
```
Comment (jesterhazy) [inline src/sagemaker_containers/_env.py]: Defaults are or are not. There is no should.

File: test/unit/test_files.py [hunk 6]
```diff
@@ -58,6 +58,17 @@ def test_read_json_throws_exception():
         _files.read_json('non-existent.json')
 
 
+def test_read_file():
+    test.write_json('test', _env.hyperparameters_file_dir)
+
+    assert _files.read_file(_env.hyperparameters_file_dir) == '\"test\"'
+
+
+def test_read_file_throws_exception():
```
Comment (jesterhazy) [inline test/unit/test_files.py]: test seems redundant - testing std lib feature

**Comment by jesterhazy** [comment]
appears to be partial duplicate of #131

**Comment by ChoiByungWook** [comment]
#131 is for container support, the older version, which currently our TensorFlow containers rely on.

File: test/unit/test_files.py (hunk 6 above)
Comment (ChoiByungWook) [reply test/unit/test_files.py]: Removed.

File: src/sagemaker_containers/_env.py [hunk 7]
```diff
@@ -770,6 +770,8 @@ class ServingEnv(_Env):
                 as specified in the user-supplied SAGEMAKER_DEFAULT_INVOCATIONS_ACCEPT environment
                 variable. Otherwise, returns 'application/json' by default.
                 For example: application/json
+            port (str): Port that SageMaker will use to handle invocations and pings against the
```
Comment (jesterhazy) [inline src/sagemaker_containers/_env.py]: doesn't match new attr name

**Pull request merged**